### DISCUSSION TITLE
How to properly name files? (workflow discussion)

### DISCUSSION THREAD

**Discussion by Cantello** [jonaswinkler/paperless-ng#1039] [Support]
I've been using paperless for some time now and currently I use as a title "Correspondent - Type - Details", e.g. "Company XY Invoice fixed roof".
This is probably different for each user here but is there some general consensus on how to best name documents? As I also enter the correspondent and the type of the document in the metadata and paperless will rename files [according to my settings](https://paperless-ng.readthedocs.io/en/latest/advanced_usage.html?highlight=PAPERLESS_FILENAME_FORMAT#file-name-handling). If I only write "Details" in the document's title, will this somehow impair the ability to search for this document?

Also, different question: I recently bought a scanner (Epson DS-730N) that can create OCRed PDFs (also to the network via a VM running the windows-only capture software). Is it advisable to let the printer's software do the OCR or is Tesseract good enough? Also, dpi count and grey vs. black/white - is there an agreement on what is best for documents to be scanned by Tesseract?

**Comment by gahsul** [comment]
I name my filenames like this:

`PAPERLESS_FILENAME_FORMAT="{created_year}/{correspondent}/[{correspondent} {created_year}{created_month}{created_day} {tag_list}]{title}_ASN{asn}"`

Because I also use tagspaces as a file browser. In tagspaces you put your tags in square brackets separated by a space. In this workflow I only have to remove the "," from the taglist beforehand.

But I do this elsewhere, so as not to break Papperless.

https://www.tagspaces.org/

**Reply by Cantello** [reply]
tagspaces looks cool, although I cannot really see on how it significantly expands on paperless's capabilities - unless you use tagspaces for more than simple DMS. Do you use the basic version and is it enough for your needs?

As for the naming, does this not lead to quite long filenames, unsuitable for manual management? And what do you use as the document's title, simply what it is about as you have the correpsondent set elsewhere?

**Reply by gahsul** [reply]
Yes, you are right, Tagspaces does not extend the functionality of Paperless when it comes to pure documents. I only used Tagspaces before Paperless and I use it for other files as well. That's why I keep the conversation names for tagspaces.

And yes you get long filenames if you use a lot of tags. Even if there is an alternative with sidecar files for tagspaces, I prefer to have the information in the file name. To make them easier to synchronize between devices.

In the document title I write a hint for the document content. For example, I have an invoice from "Example Shop" for a refrigerator, then the metadata for this invoice would be this:

correspondent = "example-shop"

document_type = invoice

Title = refrigerator

If that was a gift for someone, the following tags are added:

Gift, "name-of-person"

In my last post I forgot the document type in the PAPERLESS_FILENAME_FORMAT variable here is the correct one

`PAPERLESS_FILENAME_FORMAT="{created_year}/{correspondent}/[{correspondent} {created_year}{created_month}{created_day} {document_type} {tag_list}]{title}_ASN{asn}"`

In order for the filenames with tagspaces to work correctly in Paperless, I have to make sure that no spaces are used for the correspondent and the document type.

Of course you can do without the tagspace syntax if you don't use this program, but I find the information in this order useful. In this way, the documents are still well sorted by file name in each file manager.
